src/processTargets/targets/BaseTarget.ts

```diff
@@ -0,0 +1,183 @@
+  };
+}
+
+/** Parameters supported by all target classes */
+export interface CommonTargetParameters {
+  readonly editor: TextEditor;
+  readonly isReversed: boolean;
+  readonly contentRange: Range;
+  readonly position?: Position;
+  readonly thatTarget?: Target;
+}
+
+export interface CloneWithParameters {
+  readonly position?: Position;
+  readonly thatTarget?: Target;
+}
+
+export interface BaseTargetParameters extends CommonTargetParameters {
+  readonly delimiter: string;
+}
+
+export default abstract class BaseTarget implements Target {
+  protected readonly state: BaseTargetParameters;
+
+  constructor(parameters: BaseTargetParameters) {
+    this.state = {
+      editor: parameters.editor,
+      isReversed: parameters.isReversed,
+      contentRange: parameters.contentRange,
+      position: parameters.position,
+      delimiter: parameters.delimiter,
+      thatTarget: parameters.thatTarget,
+    };
+  }
+
+  get position() {
+    return this.state.position;
+  }
+  get editor() {
+    return this.state.editor;
+  }
+  get isReversed() {
+    return this.state.isReversed;
+  }
+  get contentRemovalRange() {
+    return this.contentRange;
+  }
+  get isLine() {
+    return false;
+  }
+  get isParagraph() {
+    return false;
+  }
+  get isWeak() {
+    return false;
+  }
+
+  get thatTarget(): Target {
+    return this.state.thatTarget != null
+      ? this.state.thatTarget.thatTarget
+      : this;
+  }
+
+  get contentText(): string {
+    return this.editor.document.getText(this.contentRange);
+  }
+
+  get contentSelection(): Selection {
+    return selectionFromRange(this.isReversed, this.contentRange);
+  }
+
+  get contentRange(): Range {
+    switch (this.position) {
+      case "start":
+      case "before":
+        return new Range(
+          this.state.contentRange.start,
+          this.state.contentRange.start
+        );
+      case "end":
+      case "after":
+        return new Range(
+          this.state.contentRange.end,
+          this.state.contentRange.end
+        );
+      default:
+        return this.state.contentRange;
+    }
+  }
+
+  get delimiter(): string | undefined {
+    switch (this.position) {
+      // This it NOT a raw target. Joining with this should be done on empty delimiter.
+      case "start":
+      case "end":
+        return "";
+      default:
+        return this.state.delimiter;
+    }
+  }
+
+  get leadingDelimiterRange() {
+    const { includeDelimitersInRemoval, leadingDelimiterRange } =
+      getTokenDelimiters(this.state.editor, this.state.contentRange);
+    return includeDelimitersInRemoval || this.position != null
+      ? leadingDelimiterRange
+      : undefined;
+  }
+
+  get trailingDelimiterRange() {
+    const { includeDelimitersInRemoval, trailingDelimiterRange } =
+      getTokenDelimiters(this.state.editor, this.state.contentRange);
+    return includeDelimitersInRemoval || this.position != null
+      ? trailingDelimiterRange
+      : undefined;
```

Comment:
It seems misleading to return `undefined` when `includeDelimitersInRemoval` is `true` when there actually is a leading delimiter. I think it would be cleaner to instead have `getTokenDelimiters` just return `removalRange`.  I would make the change myself, but I'm concerned about how it will interact with `ContinuousRangeTarget`.

Reply:
The method is specifically trailing range and should return undefined if non exists/should be used. 
This does indeed have ramifications for continuous range target and is why we have to keep those ranges separate.